src/V3EmitXml.cpp

```diff
@@ -319,6 +319,53 @@ class EmitXmlFileVisitor final : public VNVisitorConst {
     ~EmitXmlFileVisitor() override = default;
 };
 
+// Visitor for Begin blocks
+// Visiting every child note of begin block and prepend instance name with begin block name
+class BeginGenerateVisitor final : public VNVisitorConst {
+private:
```

Comment:
If we only do the cells, I think this is not needed, instead add a AstBegin visitor to HierCellsXmlVisitor that simply appends the current begin name to the hierarchy, and then use that in the AstCells visitor.   e.g. roughtly

    void visit(AstBegin* nodep) {
           VL_RESTORER(m_hier);
           m_hier += nodep->prettyName() + ".";
    }